nengo_spa/tests/test_semantic_pointer.py

```diff
@@ -148,6 +148,9 @@ def test_dot(rng):
     a = SemanticPointer(next(gen)) * 1.1
     b = SemanticPointer(next(gen)) * (-1.5)
     assert np.allclose(a.dot(b), np.dot(a.v, b.v))
+    assert np.allclose(a.dot(b.v), np.dot(a.v, b.v))
+    assert np.allclose(a.dot(list(b.v)), np.dot(a.v, b.v))
+    assert np.allclose(a.dot(tuple(b.v)), np.dot(a.v, b.v))
```

Comment:
Is the `other.dot(self)` code-path tested somewhere (perhaps indirectly)? Sorry if I missed this.

Reply:
Nope. In either case, that goes to the `return np.dot(self.v, other)` code-path, since `isinstance(other, Fixed) == True` which converts `other` into an `nd.array`.

But, looking at `codecov`, the `return other.dot(self)` code-path is being hit somehow, so either I'm missing something obvious, or its being hit indirectly in some other way?

Reply:
I see, it probably gets hit by a dot product with a `spa.Network`. `SemanticPointer` should not depend on that, so it cannot be tested in that test file. Also it is pretty straight-forward, so I think it is fine without an explicit test.